### PR TITLE
Headless/bsdhrubo/super admin/login

### PR DESCRIPTION
**Pull request by bsdhrubo**
# Description

(Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

## Type of change

(Please delete options that are not relevant.)

- [x] Link to a **GH Issue** or **Jira** task here for reference.
- [ ] Bug fix
- [x] New feature

# How Has This Been Tested?

(Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

- [x] Tested by myself
- [ ] Unit test

# Developer Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Review check list:

- [x] Separation of Concerns followed (Reviewer name/NaN)
- [x] Code is in sync with existing code patterns/technologies. (Reviewer name/NaN)
- [ ] Are there any obvious optimizations that will improve performance? (Reviewer name/NaN)
- [ ] Can any logging or debugging code be removed? (Reviewer name/NaN)
- [ ] Are all data inputs checked (for the correct type, length, format, and range) and encoded? (Reviewer name/NaN)
- [ ] Are invalid parameter values handled? (Reviewer name/NaN)
- [ ] Is error handling done the correct way? (Reviewer name/NaN)
- [ ] Are all the edge cases (null, 0, negative) handled properly? (Reviewer name/NaN)

### CODE REVIEW

File: packages/headless/src/database/mongodb/super-admin/super-admin.model.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      index: true,
+      unique: true,
+    },
+    countryCode: {
+      type: String,
+    },
+    phone: {
+      type: String,
+      index: true,
+      unique: true
```
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/super-admin/super-admin.model.ts:39]: this should be optional. optional field cant be unique

File: packages/headless/src/database/mongodb/super-admin/super-admin.model.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    },
+    password: {
+      type: String,
+    },
+    isMfa: Boolean,
+    mfaType: {
+      type: [String],
+    },
+    accountType: {
+      type: String,
```
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/super-admin/super-admin.model.ts]: i recommended to make it unique. plz check my mock schema provided in teams

File: packages/headless/src/database/mongodb/super-admin/super-admin.model.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      index: true,
+      unique: true
+    },
+    password: {
+      type: String,
+    },
+    isMfa: Boolean,
+    mfaType: {
+      type: [String],
```
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/super-admin/super-admin.model.ts:47]: use enum here

File: packages/headless/src/database/mongodb/super-admin/super-admin.model.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    },
+    phone: {
+      type: String,
+      index: true,
+      unique: true
+    },
+    password: {
+      type: String,
+    },
+    isMfa: Boolean,
```
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/super-admin/super-admin.model.ts]: `isMfaEnabled` would've been more context friendly

File: packages/headless/src/modules/super-admin/service/index.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import { authConfig } from "config/auth";
+import { JwtService } from "@nestjs/jwt";
+import { AdminJwtPayload } from "src/entity/auth";
+
+@Injectable()
+export class SuperAdminService{
+    constructor(private readonly superAdminRepository: SuperAdminRepository, private readonly jwtService: JwtService){}
+
+    async superAdminCreate(body: SuperAdminSignupReq): Promise<IServiceResponse<Partial<SuperAdmin>>>{ 
+        const isExist = await this.superAdminRepository.findOne({ $or: [ {email: body.email}, {phone: body.phone} ] })
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/super-admin/service/index.ts:19]: super admin will use only email & password for login init.
phone number is an extra info just

File: packages/headless/src/modules/super-admin/service/index.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        if(isExist?.phone === body.phone){
+            return errorResponse('This phone number is already exist.', null, HttpStatus.BAD_REQUEST)
+        }
+        
+        body.password = await bcrypt.hash(body.password, authConfig.salt)
+
+        const newSuperAdmin = await this.superAdminRepository.create(body)
+        if(newSuperAdmin)
+            return successResponse(PartialType(SuperAdmin), newSuperAdmin)
+        return errorResponse('Error in create super admin', null, HttpStatus.CONFLICT)
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/super-admin/service/index.ts]: this is not good practice. use parenthesis for block separation

File: packages/headless/src/modules/super-admin/service/index.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    async superAdminLogin(body: SuperAdminLoginDto): Promise<IServiceResponse<SuperAdminLoginRes>>{
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/super-admin/service/index.ts:39]: okay. this is general login flow. but for super admin we are giving option to enable multi-factor-authentication. we need to implement that too.

File: packages/headless/src/database/mongodb/super-admin/super-admin.model.ts (hunk 4 above)
Comment (bsdhrubo) [reply packages/headless/src/database/mongodb/super-admin/super-admin.model.ts]: fixed

File: packages/headless/src/database/mongodb/super-admin/super-admin.model.ts (hunk 3 above)
Comment (bsdhrubo) [reply packages/headless/src/database/mongodb/super-admin/super-admin.model.ts:47]: fixed

File: packages/headless/src/database/mongodb/super-admin/super-admin.model.ts (hunk 2 above)
Comment (bsdhrubo) [reply packages/headless/src/database/mongodb/super-admin/super-admin.model.ts]: removed

File: packages/headless/src/modules/super-admin/service/index.ts (hunk 5 above)
Comment (bsdhrubo) [reply packages/headless/src/modules/super-admin/service/index.ts:19]: login works only with email and password not phone number.

File: packages/headless/src/modules/super-admin/service/index.ts (hunk 6 above)
Comment (bsdhrubo) [reply packages/headless/src/modules/super-admin/service/index.ts]: fixed

File: packages/headless/src/database/mongodb/super-admin/super-admin.model.ts [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      unique: true,
+      required: true
+    },
+    countryCode: {
+      type: String,
+    },
+    phone: {
+      type: String,
+      index: true,
+      unique: true
```
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/super-admin/super-admin.model.ts:39]: as phone number is optipnal, if two admin has phone: null, then unique constraint will fail. so we can rely on service layer validation. Though there's an alternate solution. partial indexing can be done, but that might not work in other dbms. so remove unique index.

File: packages/headless/src/modules/super-admin/service/index.ts (hunk 5 above)
Comment (Anjarul-Islam) [reply packages/headless/src/modules/super-admin/service/index.ts:19]: can we create two account using different email and without phone number?

Comment (bsdhrubo) [reply packages/headless/src/modules/super-admin/service/index.ts:19]: yes

File: packages/headless/src/database/mongodb/super-admin/super-admin.model.ts (hunk 8 above)
Comment (bsdhrubo) [reply packages/headless/src/database/mongodb/super-admin/super-admin.model.ts:39]: Thanks vai for pointing out this point.

File: packages/headless/src/modules/super-admin/service/index.ts (hunk 7 above)
Comment (bsdhrubo) [reply packages/headless/src/modules/super-admin/service/index.ts:39]: I'm working on mfa.
After merging this PR I will give the mfa PR.

**Pull request merged**